library/include/flow/env_name.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef env_name_hpp
+#define env_name_hpp
+
+#include <ostream>
+#include <string>
+
+#include "flow/checked_value.hpp"
+
+namespace flow {
+
+struct env_name_checker
+{
+    constexpr auto operator()() const -> std::string
```

Comment:
This clang-tidy (LLVM) doesn't seem to appreciate that the default constructor for basic_string has been `constexpr` since C++20: https://en.cppreference.com/w/cpp/string/basic_string/basic_string.